### PR TITLE
Feature/LSI 1439 SR switch to php8.1

### PR DESCRIPTION
**Pull request by jeethufoc**
Update SR Code to PHP 8.1

### CODE REVIEW

File: src/Security/Authenticator/JwtConfiguration.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+use Lcobucci\JWT\Token\Plain;
+use Lcobucci\JWT\Signer\Key\InMemory;
+use Lcobucci\JWT\Signer\Rsa\Sha256;
+use Lcobucci\JWT\Validation\Constraint\SignedWith;
+use ReflectionClass;
+
+class JwtConfiguration
+{
+    /** @var Configuration object */
+    private object $jwtConfig;
```
Comment (mianchenia) [inline src/Security/Authenticator/JwtConfiguration.php]: Please update type hint.

File: src/Security/Authenticator/JwtConfiguration.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    /** @var InMemory object */
+    private object $jwtInMemoryFilePath;
```
Comment (mianchenia) [inline src/Security/Authenticator/JwtConfiguration.php]: Please update type hint.

File: src/Security/Authenticator/JwtConfiguration.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    public function __construct(
+        string $jwtPrivateKeyPath,
+        string $jwtPassphrase,
+        string $jwtPublicKeyPath
+    ) {
+        $this->privateKeyPath = $jwtPrivateKeyPath;
+        $this->passphrase = $jwtPassphrase;
+        $this->publicKeyPath = $jwtPublicKeyPath;
+
+        $this->jwtConfig  = (new ReflectionClass(Configuration::class))->newInstanceWithoutConstructor();
```
Comment (mianchenia) [inline src/Security/Authenticator/JwtConfiguration.php]: I believe it's not the way we want to use jwt lib. It's not safe to create object with bypassing the constructor.

File: src/Security/Authenticator/JwtConfiguration.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        $this->jwtInMemoryFilePath = (new ReflectionClass(InMemory::class))->newInstanceWithoutConstructor();
```
Comment (mianchenia) [inline src/Security/Authenticator/JwtConfiguration.php]: Please look at the comment above.

File: src/Security/Authenticator/JwtConfiguration.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    public function initialise(): Configuration
+    {
+        return $this->jwtConfig->forAsymmetricSigner(
```
Comment (mianchenia) [inline src/Security/Authenticator/JwtConfiguration.php]: This is a static method. So we don't need an instance to call it. Please update the call.

File: src/Security/Authenticator/JwtConfiguration.php [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+            new Sha256(),
+            $this->jwtInMemoryFilePath->file($this->privateKeyPath, $this->passphrase),
```
Comment (mianchenia) [inline src/Security/Authenticator/JwtConfiguration.php]: This is a static method. Please update the call.

File: src/Security/Authenticator/JwtConfiguration.php [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+            $this->jwtInMemoryFilePath->file($this->publicKeyPath, $this->passphrase)
```
Comment (mianchenia) [inline src/Security/Authenticator/JwtConfiguration.php]: This is a static method. Please update the call.

File: src/Security/Authenticator/JwtConfiguration.php [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+        );
+    }
+
+    public function parseJwtCredentials(string $credentials): Plain
+    {
+        $parsedToken =  $this->jwtConfig->forUnsecuredSigner()
```
Comment (mianchenia) [inline src/Security/Authenticator/JwtConfiguration.php]: This is a static method. Please update the call.

File: src/Security/Authenticator/JwtConfiguration.php [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+            ->parser()->parse($credentials);
+        assert($parsedToken instanceof Token\Plain);
+
+        return $parsedToken;
+    }
+
+    public function verifyJwtKey(): SignedWith
+    {
+        $key = $this->jwtInMemoryFilePath->file($this->publicKeyPath);
```
Comment (mianchenia) [inline src/Security/Authenticator/JwtConfiguration.php]: This is a static method. Please update the call.

File: src/Security/Authenticator/JwtConfiguration.php (hunk 4 above)
Comment (mianchenia) [inline src/Security/Authenticator/JwtConfiguration.php]: As I see below - we don't really need to create this instances. Please remove them.

File: src/Action/Security/RefreshAccessTokenAction.php [hunk 10]
```diff
@@ -76,10 +76,13 @@ public function __construct(
         $this->accessTokenTtl = $jwtAccessTokenTtl;
     }
 
-    public function __invoke(Request $request, Token $refreshToken): Response
+    public function __invoke(Request $request, Token $refreshToken): JsonResponse
     {
         try {
-            $user = $this->userRepository->findByUsernameWithAssignments($refreshToken->getClaim('aud'));
+            assert($refreshToken instanceof Token\Plain);
```
Comment (mianchenia) [inline src/Action/Security/RefreshAccessTokenAction.php]: Assertions are primary made for debug/develop/test when you want `run or die` scenario.  For business logic is much better to use `if/else`, `throw`, etc. Please replace them.

File: src/Security/Authenticator/JwtConfiguration.php [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+            $this->jwtInMemoryFilePath->file($this->privateKeyPath, $this->passphrase),
+            $this->jwtInMemoryFilePath->file($this->publicKeyPath, $this->passphrase)
+        );
+    }
+
+    public function parseJwtCredentials(string $credentials): Plain
+    {
+        $parsedToken =  $this->jwtConfig->forUnsecuredSigner()
+            ->parser()->parse($credentials);
+        assert($parsedToken instanceof Token\Plain);
```
Comment (mianchenia) [inline src/Security/Authenticator/JwtConfiguration.php]: Assertion. Please replace.

**Review by mianchenia — CHANGES_REQUESTED** [review]
Great job! Please look at my comments. We have to make few updates.

**Comment by mianchenia** [comment]
There is an error in Jenkins pipeline. Please have a look. Maybe we could fix it.

File: src/Security/Authenticator/JwtConfiguration.php [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+            InMemory::file($this->publicKeyPath, $this->passphrase)
+        );
+    }
+
+    public function parseJwtCredentials(string $credentials): Plain
+    {
+        $parsedToken = Configuration::forUnsecuredSigner()
+            ->parser()->parse($credentials);
+
+        if ($parsedToken instanceof plain) {
```
Comment (mianchenia) [inline src/Security/Authenticator/JwtConfiguration.php]: It should be in upper case.

Comment (mianchenia) [reply src/Security/Authenticator/JwtConfiguration.php]: `Plain`

**Review by mianchenia — CHANGES_REQUESTED** [review]
Good job! Only one little notice.

**Review by mianchenia — DISMISSED** [review]

**Review by SergiiTao — APPROVED** [review]

**Review by mianchenia — APPROVED** [review]

**Comment by jeethufoc** [comment]
Another release branch created to update SR Code to PHP 8.1

**Pull request closed without merging**